deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -847,17 +850,27 @@ def post_add_stage(request, name):
     if from_stage and from_stage not in stages:
         raise Exception("Can not clone from non-existing stage!")
 
-    external_id = environs_helper.create_identifier_for_new_stage(request, name, stage)
-
-    try:
-        if from_stage:
+    if from_stage:
+        try:
+            external_id = environs_helper.create_identifier_for_new_stage(request, name, stage)
             common.clone_from_stage_name(request, name, stage, name, from_stage, description, external_id)
-        else:
-            common.create_simple_stage(request,name, stage, description, external_id)
-    except TeletraanException as detail:
+        except Exception as detail:
+            if external_id:
+                try:
+                    environs_helper.delete_nimbus_identifier(request, external_id)
+                except:
+                    message = 'Also failed to delete Nimbus identifier {}. Please verify that identifier no longer exists, Error Message: {}'.format(external_id, detail)
+                    log.error(message)
+            else:
+                message = 'Failed to create identifier for {}/{}: {}'.format(name, stage, detail)
+                messages.add_message(request, messages.ERROR, message)
+                raise detail   
```

Comment:
Should we raise exception in both cases?

Reply:
Without raising the execution continues and still creates the stage. That might not be an issue, in which case it does snot need to be raised.

Reply:
The `Exception` could be anything. We should only declare a success when all expectations met. Otherwise it will leave potential bad states that are hard to diagnose and maintain.